app/controllers/restaurants_controller.rb

```diff
@@ -13,6 +13,10 @@ def index
         render json: {restaurants: @restaurants}
       end
     end
+    if params[:category_id].present?
+      @current_category = RestaurantCategory.find(params[:category_id])
+      @current_category_restaurants = @current_category.restaurants
```

Comment:
Whats happening here?  assigning a variable to itself?

Reply:
It's looking for the restaurant category that matches to params. Then we assign the variable `@current_category_restaurants` to the restaurants belonging to that category.

Reply:
hehe, should have looked it over a litte bit. misread one of the variables. it looks so alike.